docs/src/components/ChunkedSnippet.astro

```diff
@@ -75,6 +75,87 @@ const output = await prettier.format(rawOutput, {
 const codeBlockTitle = codeChunkArray[chunkNumber].title;
 ---
 
-<CodeBlock title={codeBlockTitle}>
+<CodeBlock title={codeBlockTitle} class="chunked-snippet">
   <Code code={output} lang="ts" theme="css-variables" />
+  <span>Copied!</span>
+  <button data-copy-button type="button">
+    <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' fill='none' stroke-width='1.75'>
+      <path d='M3 19a2 2 0 0 1-1-2V2a2 2 0 0 1 1-1h13a2 2 0 0 1 2 1'/>
+      <rect x='6' y='5' width='16' height='18' rx='1.5' ry='1.5'/>
+    </svg>
+    <span class="visually-hidden">Copy</span>
+  </button>
+  <textarea>{output}</textarea>
 </CodeBlock>
+
+<style>
+  .chunked-snippet {
+    position: relative;
+  }
+
+  .chunked-snippet pre:hover ~ button {
+     opacity: 0.7;
+  }
+
+  button {
+    position: absolute;
+    top: 2.5em;
+    right: 0.25em;
+    background-color: var(--sl-color-gray-6);
+    border-radius: var(--border-radius);
+    border: 1px solid var(--sl-color-gray-5);
+    cursor: pointer;
+    padding: var(--space-2xs);
+    opacity: 0;
+    transition: opacity ease-in-out 150ms;
+  }
+
+  button:hover,
+  button.active {
+    opacity: 1;
```

Comment:
When hovered over or when active class is present, the copy button is fully opaque.